solarforecastarbiter/metrics/preprocessing.py

```diff
@@ -58,6 +62,67 @@ def apply_validation(obs_df, qfilter, handle_func):
         return validated_obs, counts
 
 
+def apply_fill(fx_data, missing_forecast, start, end, interval_length):
+    """
+    Apply fill procedure to the data from the start to end timestamps.
+
+    Parameters
+    ----------
+    fx_data : pandas.Series or pandas.DataFrame
+        Forecast data with pandas.DatetimeIndex.
+    missing_forecast : str
+        Indicates what process to use for handling missing forecasts.
+        Currently supports : 'drop', 'forward', and bool or numeric value.
+    start : pandas.Timestamp
+    end : pandas.Timestamp
+    interval_length : pandas.Timedelta
+
+    Returns
+    -------
+    pandas.Series or pandas.DataFrame
+    int : Number of values filled or dropped.
+
+    TODO
+    ----
+    - Add support for 'clearsky' fillin.
+
+    """
+    try:
+        orig_dtype = fx_data.dtype
+    except AttributeError:
+        # DataFrame must be all same dtypes
+        orig_dtype = set(fx_data.dtypes).pop()
+    missing_forecast = str(missing_forecast)
+    # Create full datetime range at resolution
+    full_dt_index = pd.date_range(start=start, end=end, freq=interval_length,
+                                  name=fx_data.index.name)
```

Comment:
Ah. yes. Going to change the arguments to pass the `datamodel.Forecast` as this should also make it easier to address #500 eventually.

Reply:
@wholmgren Okay so need your input on the `closed` argument related to the expectation of the `ReportParameters`. Does the `start` and `end` date given by the report expected to be closed?  Because if I create the `DatetimeIndex` with closed it will be missing the right most for beginning and left most for ending and those values, while closed seems more related to a property of the data and not the report and the user's expectation of the dates that they provide.

Reply:
The report start/end does not have a concept of closed. For that reason, I think the right thing to do is handle it separately for each forecast, though I can't clearly explain this at the moment. @alorenzo175 do you have an opinion on what to do here?

Reply:
I would agree that the behaviour of closed should depend on the forecast being analyzed only and not on any broader report parameters